### PR TITLE
UnitTest++ now correctly supports CMake's find_package config mode

### PR DESCRIPTION
**Pull request by dhuantes**
[CMakeLists.txt]
- Bumped cmake minimum requirement to go to 3.14 as I'm
not able to test with an older version
- Added version to project so that is evident
when looking at CMakeLists.txt
- Removed include_directories as that command
affects more than just UnitTest++ in favor
of target_include_directories.
- The target_include_directories uses the
generator expressions to do the same thing
for the BUILD_INTERFACE condition but only
affects UnitTest++.  The INSTALL_INTERFACE
ensures that when UnitTest++ is installed
client applications calling find_package
for UnitTest++ only have to add the
UnitTest++ target to the target_link_libraries
and will get the correct include path
for UnitTest++ added to their include paths.
- Added DEBUG_POSTFIX to both library and
unit test to distinguish them from each other
as they are installed into the same directory
and would otherwise overwrite one another.
- Added Versioning using write_basic_package_version_file
to the install so that a client can call
find_package(UnitTest++ 2.1 REQUIRED) and it will be
able to confirm the version.  If the version is updated
you could theoretically ahve a version 2.2, 2.3 ,etc...
and the find_package mechanism will find the correct one.
the SameMajorVersion option in that call indicates
that 2.3 is compatible with 2.1 or in other words if
find_package(UnitTest++ 2.1 REQUIRED) is called and
2.3 is installed that satisfies the condition but
if only 3.0 was installed it will fail because of
'SameMajorVersion'.
- Also added installation for the Version file.

### CODE REVIEW

**Comment by grahamreeds** [comment]
What is the minimum version that supports find_package?  That's the version
it should support.

**Comment by dhuantes** [comment]
find_package has been around a  long time but the Transitive Usage requirements (I think) were introduced in 3.0/3.1 time frame.. But just noticed that the travis-ci failed and it failed because it has 3.12.4 installed and not the 3.14 requested.   Do you think I should drop back to requiring only 3.1 or 3.12 so that the CI tests pass?

**Comment by grahamreeds** [comment]
I am always wary of bumping versions.  I recently worked on a project that
used Ubuntu 14_04 and all that entails (gcc 4.8, etc).  I am pretty sure
that version was 2.9.  It's been a while so I may be misremembering.

Never go higher than you need to.

If the CI fails due to version that reduce it.  But make sure it is the
bare minimum to support the features, but not the bare maximum to pass.

Just did a quick Google and found a page that documents using it in CMake
2.6 so do you really need to bump the version?

GR

**Comment by dhuantes** [comment]
The change leverages transitive usage requirements.  From everything I've researched the minimum version required is 3.0.  So will bump it back down and hope the travis-ci clang Release test passes.  Since I didn't change any code not sure why it would suddenly start failing only that test.  We'll see.  Thanks.

**Comment by dhuantes** [comment]
All test pass now...  Acceptable for merging?

**Comment by pjohnmeyer** [comment]
@dhuantes Probably, will review at my earliest opportunity.

**Comment by dhuantes** [comment]
@pjohnmeyer Have you had a chance to look at this yet?  Just curious.  Thank you.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-cmake_minimum_required(VERSION 2.8.1)
-project(UnitTest++)
+cmake_minimum_required(VERSION 3.0)
+project(UnitTest++ VERSION 2.1)
```
Comment (pjohnmeyer) [inline CMakeLists.txt]: Let's make the version explicitly 3 parts. I agree that changing the CMake requirement merits a minor version bump, though, not a patch bump, so `2.1.0`:

```suggestion
project(UnitTest++ VERSION 2.1.0)
```

I'm not aware of CMake supporting `-pre` or `-beta` suffixes to distinguish this from a release version but I haven't been active with CMake in years.

Comment (dhuantes) [reply CMakeLists.txt]: Thanks.  Just made and pushed the changes requested.

**Comment by dhuantes** [comment]
@pjohnmeyer Just coming back to a project that uses unittest-cpp and noted this pull request had not yet been merged.  Seems like it's ready to go.  Any chance of making that happen?  Thanks.

**Comment by pjohnmeyer** [comment]
@dhuantes yup, merging now. Sorry for the delay -- I was laid off recently for COVID-related reasons and I am in the middle of an unplanned job search.

**Pull request merged**